src/ScottPlot/Plottable/RadialGaugePlot.cs

```diff
@@ -0,0 +1,709 @@
+                if (isBelow)
+                    gfx.RotateTransform((float)(RadToDeg * (theta - Math.PI / 2)));
+                else
+                    gfx.RotateTransform((float)(RadToDeg * (theta + Math.PI / 2)));
+
+                gfx.TranslateTransform((float)x, (float)y, System.Drawing.Drawing2D.MatrixOrder.Append);
+                gfx.DrawString(text[i].ToString(), font, brush, 0, 0, string_format);
+                gfx.ResetTransform();
+
+                // Increment theta the remaining half character.
+                theta -= (isBelow ? 1 : -1) * rects[i].Width / 2 * width_to_angle;
+            }
+        }
+
+
+        /// <summary>
+        /// Measure the characters in a string with no more than 32 characters.
+        /// </summary>
+        /// <param name="gfx"></param>
+        /// <param name="font"></param>
+        /// <param name="clientRectangle"></param>
+        /// <param name="text"></param>
+        /// <returns></returns>
+        private List<RectangleF> MeasureCharactersInWord(Graphics gfx, System.Drawing.Font font, RectangleF clientRectangle, string text)
+        {
+            List<RectangleF> result = new();
+
+            using StringFormat string_format = new();
+
+            string_format.Alignment = StringAlignment.Center;
+            string_format.LineAlignment = StringAlignment.Center;
+            string_format.Trimming = StringTrimming.None;
+            string_format.FormatFlags = StringFormatFlags.MeasureTrailingSpaces;
+
+            CharacterRange[] ranges = new CharacterRange[text.Length];
+            for (int i = 0; i < text.Length; i++)
+            {
+                ranges[i] = new CharacterRange(i, 1);
+            }
+            string_format.SetMeasurableCharacterRanges(ranges);
+
+            // Find the character ranges.
+            RectangleF rect = new RectangleF(0, 0, 10000, 100);
+            Region[] regions =
+                gfx.MeasureCharacterRanges(
+                    text, font, clientRectangle,
+                    string_format);
+
+            // Convert the regions into rectangles.
+            foreach (Region region in regions)
+                result.Add(region.GetBounds(gfx));
+
+            return result;
+        }
+
+        /// <summary>
+        /// Measure the characters in the string.
+        /// </summary>
+        /// <param name="gfx"></param>
+        /// <param name="font"></param>
+        /// <param name="clientRectangle"></param>
+        /// <param name="text"></param>
+        /// <returns></returns>
+        private List<RectangleF> MeasureCharacters(Graphics gfx, System.Drawing.Font font, RectangleF clientRectangle, string text)
+        {
+            List<RectangleF> results = new List<RectangleF>();
+
+            // The X location for the next character.
+            float x = 0;
+
+            // Get the character sizes 31 characters at a time.
+            for (int start = 0; start < text.Length; start += 32)
```

Comment:
Why is this in batches of 32?

Reply:
I believe that's because [SetMeasurableCharacterRanges](https://docs.microsoft.com/es-es/dotnet/api/system.drawing.stringformat.setmeasurablecharacterranges?view=net-5.0), which is called a few lines later inside
https://github.com/ScottPlot/ScottPlot/blob/d7f90e1346e91c60c85a45e2d2f75b36302bd19a/src/ScottPlot/Plottable/RadialGaugePlot.cs#L684
throws an `OveflowException` when the array passed is longer than 32 items.

Reply:
I may refactor this by moving some functionality into the `ScottPlot.Drawing` module and improve it along the way.

Overall there is a lot of code here to measure/render text character-by-character along a curve. Moving that logic out of this plot type will make it more testable and also simplify the code in this module 👍